jib-core/src/main/java/com/google/cloud/tools/jib/http/FailoverHttpClient.java

```diff
@@ -209,10 +221,18 @@ public Response put(URL url, Request request) throws IOException {
    * @throws IOException if building the HTTP request fails.
    */
   public Response call(String httpMethod, URL url, Request request) throws IOException {
-    if (!isHttpsProtocol(url) && !enableHttpAndInsecureFailover) {
+    if (!isHttpsProtocol(url)) {
+      if (enableHttpAndInsecureFailover) {
```

Comment:
I find this part a little confusing: It seems like we're checking for: "http and insecure https are not allowed", but really we're only checking for "http not allowed"?

Reply:
The `enableHttpAndInsecureFailover` really indicates two things shoved into one flag (ANDed): `enableHttp` AND `enableInsecureFailover`. We can actually split this into two flags, but I thought this isn't really worth it, especially in our context.

And this was actually my attempt to make the code easier to understand by putting a short-circuit (kind of an optimization); I don't really have to change this line in terms of net effect, and things would work as intended without this change actually. This short-circuit (optimization) is like this:

- If the request was plain-HTTP, then
   - if HTTP is allowed, just call the endpoint right away and exit. No need to think about failover.
   - if not allowed, fail right away with `SSLException`.

Does this make sense? Does the code need some more tweaks to make it less confusing?

Reply:
Right, I think this is fine.
A comment next to it (`// we only care if http is enabled`) or something might be helpful but isn't necessary.